### PR TITLE
Servers migrated to Jakarta 

### PR DESCRIPTION
**Pull request by jadelkhoury**
All 4 refImpl servers generated based on the latest LyoDesigner (currently under PR https://github.com/eclipse/lyo.designer/pull/276) to support migration to Jakarta.

### CODE REVIEW

File: src/server-rm/pom.xml [hunk 1]
```diff
@@ -106,6 +106,23 @@
...
             </build>
         </profile>
+	    <profile>
+			<!-- some servlet containers (Tomcat) does not ship with a JSTL impl
+			In such cases, run with this profile-->
+	        <id>with-jstl-impl</id>
+	        <dependencies>
+				<dependency>
+		            <groupId>jakarta.servlet.jsp.jstl</groupId>
+		            <artifactId>jakarta.servlet.jsp.jstl-api</artifactId>
```
Comment (berezovskyi) [inline src/server-rm/pom.xml]: Is this needed? I think the impl below will ensure api classes are present.

File: src/server-rm/pom.xml [hunk 2]
```diff
@@ -106,7 +106,34 @@
             </plugins>
             </build>
         </profile>
+        <!-- End of user code
+        -->
+        <profile>
+        <!-- some servlet containers (Tomcat) does not ship with a JSTL impl
+        In such cases, run with this profile-->
+            <id>with-jstl-impl</id>
```
Comment (berezovskyi) [inline src/server-rm/pom.xml:114]: please update GH Action workflows to use this profile when testing with tomcat.

**Review by berezovskyi — CHANGES_REQUESTED** [review]
Looks good, need corresponding -P change to the CI workflows and that's about it.

**Comment by berezovskyi** [comment]
https://github.com/oslc-op/refimpl/actions/runs/8498710257

File: src/server-rm/pom.xml (hunk 1 above)
Comment (Jad-el-khoury) [reply src/server-rm/pom.xml]: It is needed. Don't ask my why, but when I remove "jakarta.servlet.jsp.jstl-api" (but keep the impl, I get the following at runtime when trying to get a jsp page.

[INFO] Apr 01, 2024 3:14:40 PM org.apache.catalina.core.ApplicationDispatcher invoke
[INFO] SEVERE: Servlet.service() for servlet [jsp] threw exception
[INFO] java.lang.ClassNotFoundException: jakarta.servlet.jsp.jstl.core.ConditionalTagSupport
[INFO] 	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1353)
[INFO] 	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1165)
[INFO] 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
[INFO] 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
[INFO] 	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)

**Comment by berezovskyi** [comment]
I think it's ready to merge. Do you think you could please resolve merge conflicts?

**Comment by Jad-el-khoury** [comment]
@berezovskyi I can try. But there is still 1 change request still pending on your side?

**Comment by jadelkhoury** [comment]
and, did you see my latest changes to the readme file?

**Comment by Jad-el-khoury** [comment]
@berezovskyi ! We should first merge Lyo, before we merge this PR. Ready to go there?

**Comment by berezovskyi** [comment]
> and, did you see my latest changes to the readme file?

yes, looks good



> But there is still 1 change request still pending on your side?

ah, forgot that. btw, only RM was "tomcatified", we need 3 more docker files and to update the compose yaml.

**Comment by jadelkhoury** [comment]
> ah, forgot that. btw, only RM was "tomcatified", we need 3 more docker files and to update the compose yaml.

Sorry! Don't get what you are after here.

**Comment by berezovskyi** [comment]
I think the build will succeed only once Jenkins pushes new SNAPSHOTS to eclipse repos.

**Comment by berezovskyi** [comment]
> Sorry! Don't get what you are after here.

I can take a look at that.

**Comment by berezovskyi** [comment]
> I can then leave it to you to merge the PR also then?

I think your check will be needed on `src/lib-common/pom.xml`, it was referencing `6.0.0.Alpha1`. Also the kotlin client is broken.

**Comment by Jad-el-khoury** [comment]
THat was the conflict I had in the other pom.xml files. For some reason, a merge has also changed the Lyo version to Alpha. 
src/lib-common/pom.xml looks good.

**Pull request merged**